src/auto-aspect-executor.ts

```diff
@@ -45,33 +48,26 @@ export class AutoAspectExecutor implements OnModuleInit {
             originalFn: any;
             metadata?: unknown;
             aopSymbol: symbol;
-          }[] = this.reflector.get(metadataKey, instance[methodName]);
+          }[] = this.reflector.get(metadataKey, target[methodName]);
           if (!metadataList) {
             return;
           }
 
-          // TODO: Support request scope providers
-          if (!wrapper.isDependencyTreeStatic()) {
-            console.warn(
-              `[${
-                wrapper.instance?.constructor?.name || 'Unknown'
-              }] "@tossteam/nestjs-aop" does not yet support request scope providers`,
-            );
-            return;
-          }
-
           for (const { originalFn, metadata, aopSymbol } of metadataList) {
-            const wrappedMethod = lazyDecorator.wrap({
-              instance,
-              methodName,
-              method: originalFn.bind(instance),
-              metadata,
+            const proxy = new Proxy(target[methodName], {
+              apply: (_, thisArg, args) => {
+                const wrappedMethod = lazyDecorator.wrap({
+                  instance: thisArg,
+                  methodName,
+                  method: originalFn.bind(thisArg),
+                  metadata,
+                });
```

Comment:
e.g.
```suggestion 
                const cached = weakMap.get(thisArg) // + using aopSymbol, methodName for key
                if (cached) return cached;
                const wrappedMethod = lazyDecorator.wrap({
                  instance: thisArg,
                  methodName,
                  method: originalFn.bind(thisArg),
                  metadata,
                });
                weakMap.set(thisArg /* + aopSymbol and methodName*/, wrappedMethod)
```

Reply:
Great idea! I'll apply caching like below, how do you think? In my local test, it works well and passed all tests. Checking decorated functions use cached value also done.

```ts
private readonly wrappedMethodCache = new WeakMap(); // on class field
...
...
const cached = this.wrappedMethodCache.get(thisArg);
if (cached?.[aopSymbol]?.[methodName]) {
  return Reflect.apply(cached[aopSymbol][methodName], lazyDecorator, args);
}
const wrappedMethod = lazyDecorator.wrap({
  instance: thisArg,
  methodName,
  method: originalFn.bind(thisArg),
  metadata,
});
this.wrappedMethodCache.set(thisArg, {
  ...cached,
  [aopSymbol]: {
    ...cached?.[aopSymbol],
    [methodName]: wrappedMethod,
  },
});
```

Reply:
Sounds good. Personally, I think this is a little easier to read. How about going like this?
```suggestion
const cached = this.wrappedMethodCache.get(thisArg) || {};
if (cached[aopSymbol]?.[methodName]) {
  return Reflect.apply(cached[aopSymbol][methodName], lazyDecorator, args);
}
const wrappedMethod = lazyDecorator.wrap({
  instance: thisArg,
  methodName,
  method: originalFn.bind(thisArg),
  metadata,
});
cached[aopSymbol] ??= [];
cached[aopSymbol][methodName] = wrappedMethod;
this.wrappedMethodCache.set(thisArg, cached);
```

Reply:
It looks better. I'll add a commit with your suggestion.